lib/devise_user_metering/model.rb

```diff
@@ -4,14 +4,33 @@ module Devise
   module Models
     module UserMetering
       
-      #this function is used to calculate the standard activity for the month of the time passed in
+      # takes a time
+      # returns a decimal between 0 and 1 that reflects the proportion of time in the given month
+      # that the user has been 'active'
       def active_proportion_of_month(time)
         active_proportion(time, time.beginning_of_month, time.end_of_month)
       end
       
       #this function is used to calculate the standard activity for the custom interval passed in
-      def active_proportion_of_interval(time, time_start, time_end)
-        active_proportion(time, time_start, time_end)
+      def active_proportion_of_interval(time, interval_start, interval_end)
+        if interval_end > Time.now
+          raise StandardError.new("You can't get meter data for partial intervals")
+        end
+        if interval_start < self.activated_at
```

Comment:
This used to be compared to `self.activated_at.beginning_of_month` now that the interval isn't based on months I want to determine any side effects removing the "beginning_of_month" part has